packages/react-aria-components/docs/advanced.mdx

```diff
@@ -0,0 +1,345 @@
+
+```tsx
+<NumberField>
+  <Label>Width</Label>
+  <Group>
+    <Input />
+    <Button slot="increment">+</Button>
+    <Button slot="decrement">-</Button>
+  </Group>
+</NumberField>
+```
+
+React Aria Components automatically provide behavior to their children by passing event handlers and other attributes via context. For example, the increment and decrement buttons in a `NumberField` receive `onPress` handlers that update the value. Keeping each element of a component separate enables full styling, layout, and DOM structure control, and contexts ensure that accessibility and behavior are taken care of on your behalf.
+
+This architecture also enables you to reuse React Aria Components in your own custom patterns, or even replace one part of a component with your own custom implementation without rebuilding the whole pattern from scratch.
+
+### Custom patterns
+
+Each React Aria Component exports a corresponding context that you can use to build your own compositional APIs similar to the builtin components. You can send any prop or ref via context that you could pass to the corresponding component. The local props and ref on the component are merged with the ones passed via context, with the local props taking precedence (following the rules documented in [mergeProps](mergeProps.html)).
+
+This example shows a `FieldGroup` component that renders a group of text fields. The entire group can be marked as disabled via the `isDisabled` prop, which is passed to all child text fields via the `TextFieldContext` provider.
+
+```tsx
+import {TextFieldContext} from 'react-aria-components';
+
+interface FieldGroupProps {
+  children?: React.ReactNode,
+  isDisabled?: boolean
+}
+
+function FieldGroup({children, isDisabled}: FieldGroupProps) {
+  return (
+    /*- begin highlight -*/
+    <TextFieldContext.Provider value={{isDisabled}}>
+    {/*- end highlight -*/}
+      {children}
+    </TextFieldContext.Provider>
+  );
+}
+```
+
+Any `TextField` component you place inside a `FieldGroup` will automatically receive the `isDisabled` prop from the group, including those that are deeply nested inside other components.
+
+```tsx
+<FieldGroup isDisabled={isSubmitting}>
+  <MyTextField label="Name" />
+  <MyTextField label="Email" />
+  <CreditCardFields />
+</FieldGroup>
+```
+
+The contexts consumed by each component are listed in the Advanced Customization section of their documentation.
+
+### Slots
+
+Some patterns include multiple instances of the same component. These use the `slot` prop to distinguish each instance. Separate props can be sent to each slot by providing an object with keys for each slot name.
```

Comment:
Do we need to explain this concept? I know we have a page already https://react-spectrum.adobe.com/react-spectrum/layout.html#slots but its in RSP, is that enough?